### ISSUE TITLE
Issue with gcc 7.2.1 

### ISSUE THREAD

**Issue by rmcxm** [KlausT/ccminer#129]
I am on fedora 27 and gcc is unable to be downgraded. 
If I remove and install gcc 4.x from tar it gives me errors.
So I am compiliing with gcc 7.2.1 and i have disabled gcc version check in file /usr/local/cuda/include/crt/host_config.h 
Also disabled unnecessary arch in Makefile.am

When I run build.sh, I am getting below error 

```
/usr/local/cuda/bin/nvcc -gencode=arch=compute_50,code=sm_50 -I/usr/local/cuda/include -I. -O3 -std=c++11 -Xcompiler -fno-strict-aliasing,-Wall -D_FORCE_INLINES  --ptxas-options="-v" --maxrregcount=128 -o Algo256/cuda_fugue256.o -c Algo256/cuda_fugue256.cu
/usr/include/bits/floatn.h(61): error: invalid argument to attribute "__mode__"

/usr/include/bits/floatn.h(73): error: identifier "__float128" is undefined

2 errors detected in the compilation of "/tmp/tmpxft_00004b31_00000000-6_cuda_fugue256.cpp1.ii".
```

Please help!

**Comment by corsel** [comment]
I have the same problem in OpenSUSE, even though I installed gcc-6.4.0 to mitigate the compatibility issues with symlinks to old gcc in cuda's bin and all. Have you been able to solve the issue?

**Comment by rmcxm** [comment]
i moved to tpruvot version of ccminer. also we don't have to downgrade. just forcefully compile and make install using steps at [https://github.com/tpruvot/ccminer/blob/windows/INSTALL](url)


# Step 4: Fix the toolkit incompatibility with gcc 6

# You need to build yourself an older GCC/G++ version, i recommend the 5.4
# see https://gcc.gnu.org/mirrors.html
# Note: this manual method will override the default gcc, it could be better to use a custom toolchain prefix

wget ftp://ftp.lip6.fr/pub/gcc/releases/gcc-5.4.0/gcc-5.4.0.tar.bz2
dnf install libmpc-devel mpfr-devel gmp-devel
./configure --prefix=/usr/local --enable-languages=c,c++,lto --disable-multilib
make -j 8 && make install
(while this step, you have the time to cook something :p)

**Comment by corsel** [comment]
Yep, I've already done this. Compiled gcc 6.4.0 from scratch into /opt and symlinked it into cuda's bin (and I had a good night's sleep in make step instead of cooking :)) Some other issues arised, where I had to apply some patches to gcc headers (certain typedefs are problematic, they've been fixed in gcc 7 but you know, cuda just cannot do with 7), exclude some libraries in ./configure step (and get another good night's sleep in the meantime). Maybe I'm somehow failing to make cuda use that version, instead of my regular new gcc. I'll post again if I find a way.
